Source/Filters/Trim/YPTimeStampScrollView.swift

```diff
@@ -31,41 +48,83 @@ class YPTimeStampScrollableView: UIScrollView {
+            }
+        }
+        return timeStamps
+    }
+
+    func generateTimeRanges(for assetDuration: CMTime) -> [YPTimeStampViewModel] {
+        let totalSeconds = CMTimeGetSeconds(assetDuration)
+        var timeStamps: [YPTimeStampViewModel] = []
+
+        var timeRangeAmount: Double = 0
+        if totalSeconds < 30 {
+            // Clips less then 30 seconds will have a scale of 1 dot = 1 second and every 5th dot will render a timestamp label.
+            timeStamps = generateTimeStampViewModels(amount: Int(totalSeconds), timeStampInterval: 5, duration: 1)
+        } else if totalSeconds >= 30, totalSeconds <= 61 {
+            let fiveSecondDivisor: Double = 5.0
+            timeRangeAmount = round(totalSeconds / fiveSecondDivisor)
+            // Clips between 30 seconds and 61 seconds will have a scale of 1 dot = 5 seconds and every 3rd dot will render a timestamp label.
+            timeStamps = generateTimeStampViewModels(amount: Int(timeRangeAmount), timeStampInterval: 3, duration: 5)
+        } else {
+            let tenSecondDivisor: Double = 10
+            timeRangeAmount = round(totalSeconds / tenSecondDivisor)
+            // Clips greater then 61 seconds will have a scale of 1 dot = 10 seconds and every 3rd dot will render a timestmp label.
+            timeStamps = generateTimeStampViewModels(amount: Int(timeRangeAmount), timeStampInterval: 3, duration: 10)
+        }
+        return timeStamps
+    }
+
+    func renderRanges() {
+        guard let videoDuration = asset?.duration else { return }
 
-        var viewCount = Int(contentSize.width) / width
-        // The viewCount is adjusted here to account for the sliding trim handle views. These views consume horizontal space on the
-        // left and the right sides of the trimmer view, and the contentSize of the trimmer view does not extend before the left sliding view or extend
-        // beyond the right sliding view.
-        viewCount += 2
+        let ranges = generateTimeRanges(for: videoDuration)
+        let subViewWidth = contentSize.width / CGFloat(ranges.count)
+        let handleBarWidth: CGFloat = 15
 
-        for i in 0..<viewCount{
-            let contentViewSubView = YPTrimmerTimeStampView(
+        for i in 0..<ranges.count {
+            let range = ranges[i]
+            let contentViewSubView = YPTimeStampView(
                 timeStampFont: timeStampFont,
                 timeStampColor: timeStampColor,
-                timeBarColor: timeBarColor
+                timeBarColor: range.shouldRenderTimeStamp ? timeBarLargeCircleColor : timeBarSmallCircleColor
             )
 
-            let xPosition = i * width
-
-            if i == 0, let asset = asset {
-                contentViewSubView.shouldRenderBoldCircle = true
-                contentViewSubView.timeStampText = getTime(from: CGFloat(xPosition), for: asset)?.durationText
-            }
+            guard var xPosition = getPosition(from: range.startTime) else { continue }
```

Comment:
Should we consider logging if the guard returns here? Wondering if returning causes the UI to break.